File: Python3/Remove Element.py
# 36ms, 100.0%
class Solution:
    def removeElement(self, nums, val):
        count, length = 0, len(nums)
        for i in range(length):
            if nums[i - count] == val:
                nums[i - count: i - count + 1] = []
                count += 1
        return length - count
        

File: Java/Remove Element.java
// 0ms, 100.00%; 38.1MB, 5.05%
class Solution {
    public int removeElement(int[] nums, int val) {
        int length = 0;
        for (int i = 0; i < nums.length; ++i) {
            if (nums[i] != val) {
                nums[length++] = nums[i];
            }
        }
        return length;
    }
}
